src/subcommand/wallet/inscribe.rs

```diff
@@ -62,37 +65,60 @@ impl Inscribe {
         inscription,
         inscriptions,
         options.chain().network(),
-        utxos,
+        utxos.clone(),
         commit_tx_change,
         reveal_tx_destination,
         self.fee_rate,
       )?;
 
-    if !self.no_backup {
-      Inscribe::backup_recovery_key(&client, recovery_key_pair, options.chain().network())?;
-    }
+    let mut all_utxos = utxos.clone();
+    all_utxos.insert(
+      reveal_tx.input[0].previous_output,
+      Amount::from_sat(unsigned_commit_tx.output[0].value),
+    );
```

Comment:
May also need `.cloned` call or something here after iter:

```suggestion
    let utxos_with_commit_tx_output = utxos.iter().chain(
       // use commit tx ouput (must construct outpoint)
      Amount::from_sat(unsigned_commit_tx.output[0].value),
    ).collect();
 ```